src/Pos/Explorer/Web/Server.hs

```diff
@@ -586,15 +580,38 @@ topsortTxsOrFail f =
     maybeThrow (Internal "Dependency loop in txs set") .
     topsortTxs f
 
--- | Convert server address to client address,
--- with the possible exception (effect).
+-- | Deserialize Cardano or RSCoin address and convert it to Cardano address.
+-- Throw exception on failure.
 cAddrToAddr :: MonadThrow m => CAddress -> m Address
-cAddrToAddr cAddr = either exception pure (fromCAddress cAddr)
+cAddrToAddr cAddr@(CAddress rawAddrText) =
+    if (not $ T.null rawAddrText) && (T.last rawAddrText == '=') then do
```

Comment:
I guess you are testing here is valid base64. Maybe better way would be to try to run base64 on that string and see if there will be any errors. Look here how it is done in redeemAda endpoint https://github.com/input-output-hk/cardano-sl/blob/master/src/Pos/Wallet/Web/Server/Methods.hs#L1087 . We are also checking there that it can be both:
 * valid base64
 * valid base64 url

There are anecdotal problems with avvm where we used both base64 and base64 url. 

That might be the problem that you are running at maybe with CSM-181. Take into account that rscoin address can be in either valid base64 or base64url format.